main/main.cpp

```diff
@@ -1013,6 +1014,9 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 
 			quiet_stdout = true;
 
+		} else if (I->get() == "--no-header") {
+			Engine::singleton->_print_header = false;
```

Comment:
Both seem to work, but below you're using this form, it would be best to keep it consistent:
```suggestion
			Engine::get_singleton()->_print_header = false;
```